js/ui/appDisplay.js

```diff
@@ -2382,7 +2401,10 @@ const AppIcon = new Lang.Class({
         return [sourceActor.actor];
     },
 
-    _onNewGtkNotificationSource: function(daemon, source) {
+    _sourceAdded: function(tray, source) {
+        if (!(source instanceof NotificationDaemon.GtkNotificationDaemonAppSource))
+            return;
```

Comment:
Hmm okay. Can you add a comment specifying this? It might be confusing for future maintainers if they see this.